lib/llm/src/block_manager/connector/scheduler.rs

```diff
@@ -488,35 +503,86 @@ impl Scheduler {
     fn handle_schedule_request(&mut self, request: TransferScheduleRequest) {
         let request_id = request.leader_request.request_id.clone();
 
-        // the slot may or may not exist
-        // if it exists, the worker side may or may not have arrived first
-
-        // tracing::debug!("engine state adding slot");
-        // let slot = SchedulerSlot::new(request, self.iteration);
-        // self.slots.insert(request_id, slot);
-
-        // capture a clone of the atomic counter, and pass it through with the request
+        // Check if there's already an enqueued request from the worker for this operation
+        if let Some(worker_requests) = self.enqueued_requests.get(&request_id) {
+            if worker_requests.contains_key(&request.leader_request.uuid) {
+                // Worker arrived first, now we have both sides - execute the transfer
+                self.schedule_request(request);
+                return;
+            }
+        }
 
-        unimplemented!("@ziqi")
+        // Transfer arrived first, store it for when the worker arrives
+        // TODO(ziqif): should we remove from enqueued_requests at all when the request is completed?
```

Comment:
yep.  when both requests are present, we can remove the uuid entry and move the object from the wait phase to the active queue.  in our case, the active queue, immediately spawns the task to execute the release, awaits the response and inc the counter.

on slot deletions we should 

```
if let Some(inner) = self.enqueued_requests.remove(&request_id) {
    debug_assert!(inner.is_empty());
}